sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -121,20 +122,28 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         try {
             final SwaggerMethodParser methodParser = getMethodParser(method);
             final HttpRequest request = createHttpRequest(methodParser, args);
-            Context context = methodParser.setContext(args)
-                .addData("caller-method", methodParser.getFullyQualifiedMethodName())
+            Context context = methodParser.setContext(args);
+
+            RequestOptions options = methodParser.setRequestOptions(args);
+            if (options != null) {
+                options.getRequestCallback().accept(request);
```

Comment:
This `accept` is now before `request.setBody(validateLength(request));`, will it be a problem?

Reply:
Hmm, that's a good question. For safety, given how late this change is going in, I'll have two if blocks to use the `RequestOptions`.